internal/app/machined/internal/api/reg/reg.go

```diff
@@ -507,3 +508,32 @@ func getContainerInspector(ctx context.Context, namespace string, driver common.
 		return nil, fmt.Errorf("unsupported driver %q", driver)
 	}
 }
+
+// Read implements the read API.
+func (r *Registrator) Read(ctx context.Context, in *machineapi.ReadRequest) (reply *common.DataReply, err error) {
+	f, err := os.Stat(in.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	switch mode := f.Mode(); {
+	case mode.IsRegular():
+		var b []byte
+
+		if b, err = ioutil.ReadFile(in.Path); err != nil {
+			return nil, err
+		}
+
+		reply = &common.DataReply{
+			Response: []*common.DataResponse{
+				{
+					Bytes: &common.Data{Bytes: b},
+				},
+			},
+		}
+
+		return reply, nil
+	default:
+		return nil, fmt.Errorf("path must be a file")
```

Comment:
"regular file", perhaps?  As it stands, things like device nodes, sockets, etc what _are_ files will still get caught by this filter.

All of these are caught by this filter:
```
ModeType = ModeDir | ModeSymlink | ModeNamedPipe | ModeSocket | ModeDevice | ModeCharDevice | ModeIrregular
```

Reply:
Fixed